bigtable-hbase-parent/bigtable-hbase-1.x-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestFilters.java

```diff
@@ -95,6 +96,43 @@ public void testBigtableFilter() throws IOException {
       Assert.assertTrue(CellUtil.matchingValue(result.rawCells()[0], valA));
     }
   }
+  
+  /**
+   * This test case is used to validate TimestampRangeFilter with Integer.MAX_VALUE #1552
+   * 
+   * @throws IOException
+   */
+  @Test
+  public void testTimestampRangeFilterWithMaxVal() throws IOException {
+	    // Initialize
+	    int numCols = Integer.MAX_VALUE;
+	    Long start = (long) (Integer.MAX_VALUE - 2);
+	    String goodValue = "includeThisValue";
+	    Table table = getDefaultTable();
+	    byte[] rowKey = dataHelper.randomData("testRow-TimestampRange-");
+	    Put put = new Put(rowKey);
+	    for (Long i = start; i < numCols; ++i) {
+	      put.addColumn(COLUMN_FAMILY, dataHelper.randomData(""), i, Bytes.toBytes(goodValue));
+	    }
+	    table.put(put);
+
+	    // Filter for results
+	    FilterList filter = new FilterList(FilterList.Operator.MUST_PASS_ALL,
+				  new TimestampRangeFilter(start, Integer.MAX_VALUE),
+				  new FamilyFilter(CompareOp.EQUAL, new BinaryComparator(Bytes.toBytes("test_family"))));
```

Comment:
Please only use the `TimestampRangeFilter` and not the `FilterList` for clarity.